client/src/unifyfs-sysio.c

```diff
@@ -246,7 +246,7 @@ static char* _getcwd_impl(char* path, size_t size)
          * that is big enough */
         buf = (char*) malloc(len);
         if (buf != NULL) {
-            strncpy(buf, unifyfs_cwd, len);
+            memcpy(buf, unifyfs_cwd, strlen(unifyfs_cwd));
```

Comment:
This was the fix used to patch our Spack package in https://github.com/spack/spack/pull/22066.
Want to check that this is the solution we want to use here.

Reply:
It is fine to use memcpy, but you need to use `strlen(string) + 1` for the length to make sure the NUL character is included. `strncpy` makes sure the string is NUL-terminated implicitly.

Reply:
Updated.

Reply:
Yep, that looks good.  Should we use ``strlcpy`` here to be consistent?

It's bigger than this PR, but grep turns up a bunch of other ``strncpy`` hits.  Maybe it's worth doing a full sweep of the code base to review a full replacement of ``strncpy`` with ``strlcpy``.

Reply:
I'm happy to change this one to `strlcpy` for consistency and we can tackle the others in another PR. These were the only two the new compilers were currently complaining about.